src/main/java/org/breedinginsight/brapps/importer/services/processors/ExperimentProcessor.java

```diff
@@ -293,14 +317,32 @@ private void getNewBrapiData(List<BrAPIImport> importRows, Program program, User
             PendingImportObject<BrAPIObservationUnit> obsUnitPIO = createObsUnitPIO(program, commit, envSeqValue, importRow);
             String key = createObservationUnitKey(importRow);
             this.observationUnitByNameNoScope.put(key, obsUnitPIO);
+
+            for (Column<?> column : phenotypeCols) {
+                PendingImportObject<BrAPIObservation> obsPIO = createObservationPIO(importRow, column.name(), column.getString(i));
+                this.observationByHash.put(getImportObservationHash(importRow, getVariableNameFromColumn(column)), obsPIO);
+            }
         }
     }
 
-    private String createObservationUnitKey(ExperimentObservation importRow) {
+    private static String createObservationUnitKey(ExperimentObservation importRow) {
         String key = importRow.getEnv() + importRow.getExpUnitId();
         return key;
     }
 
+    private static String getImportObservationHash(ExperimentObservation importRow, String variableName) {
+        // TODO: handle timestamps once we support them
+        return getObservationHash(createObservationUnitKey(importRow), variableName, importRow.getEnv());
+    }
+
+    //TODO: Add timestamp parameter once we support them
+    private static String getObservationHash(String observationUnitName, String variableName, String studyName) {
+        String concat = DigestUtils.sha256Hex(observationUnitName) +
+                DigestUtils.sha256Hex(variableName) +
+                DigestUtils.sha256Hex(studyName);
+        return DigestUtils.sha256Hex(concat);
+    }
+
```

Comment:
Why are these methods static?

Reply:
I had them as static in the old ObservationProcessor and just brought that over because there is no dependence on the class instance.

Reply:
Pushed changes.